src/renderer/InvestJob.js

```diff
@@ -43,83 +43,82 @@ export default class InvestJob {
     return InvestJob.getJobStore();
   }
 
-  /**
-   * @param {object} obj - the metadata property
-   * @param {string} obj.modelRunName - name to be passed to `invest run`
-   * @param {string} obj.modelHumanName - colloquial name of the invest model
-   * @param {object} obj.argsValues - an invest "args dict" with initial values
-   * @param {string} obj.logfile - path to an existing invest logfile
-   * @param {'running'|'success'|'error'} obj.status - status of the invest process
-   */
-  constructor(
-    {
-      modelRunName,
-      modelHumanName,
-      argsValues,
-      logfile,
-      status,
-    }
-  ) {
-    this.metadata = {};
-    this.metadata.modelRunName = modelRunName;
-    this.metadata.modelHumanName = modelHumanName;
-    this.metadata.argsValues = argsValues;
-    this.metadata.logfile = logfile;
-    this.metadata.status = status;
-    this.metadata.workspaceHash = null;
-
-    this.save = this.save.bind(this);
-    this.setProperty = this.setProperty.bind(this);
-    this.setWorkspaceHash = this.setWorkspaceHash.bind(this);
-  }
-
-  setWorkspaceHash() {
-    if (this.metadata.argsValues.workspace_dir
-        && this.metadata.modelRunName) {
-      this.metadata.workspaceHash = crypto.createHash('sha1').update(
-        `${this.metadata.modelRunName}
-         ${JSON.stringify(this.metadata.argsValues.workspace_dir)}
-         ${JSON.stringify(this.metadata.argsValues.results_suffix)}`
+  static getWorkspaceHash(modelRunName, workspaceDir, resultsSuffix) {
+    if (workspaceDir && modelRunName) {
+      const workspaceHash = crypto.createHash('sha1').update(
+        `${modelRunName}
+         ${JSON.stringify(workspaceDir)}
+         ${JSON.stringify(resultsSuffix)}`
       ).digest('hex');
-    } else {
-      throw Error(
-        'Cannot hash a job that is missing workspace or modelRunName properties'
-      );
+      return workspaceHash;
     }
+    throw Error(
+      'Cannot hash a job that is missing workspace or modelRunName properties'
+    );
   }
 
-  setProperty(key, value) {
-    this.metadata[key] = value;
-  }
-
-  async save() {
-    if (!this.metadata.workspaceHash) {
-      this.setWorkspaceHash();
+  static async saveJob(job) {
+    if (!job.workspaceHash) {
+      job.workspaceHash = this.getWorkspaceHash(
+        job.modelRunName,
+        job.argsValues.workspace_dir,
+        job.argsValues.resultsSuffix
+      );
     }
     const isoDate = new Date().toISOString().split('T')[0];
     const localTime = new Date().toTimeString().split(' ')[0];
-    this.metadata.humanTime = `${isoDate}  ${localTime}`;
-    let sortedKeys = await investJobStore.getItem(HASH_ARRAY_KEY);
-    if (!sortedKeys) {
+    job.humanTime = `${isoDate} ${localTime}`;
+    let sortedWorkspaceHashes = await investJobStore.getItem(HASH_ARRAY_KEY);
```

Comment:
I noticed you switched from `sortedKeys` to `sortedWorkspaceHashes`. Would it be consistent to also use this naming above in the static functions? It's not necessary, but I thought maybe it'd be nice if we were taking the time to use a more thoughtful name.

Reply:
Good idea!